### DISCUSSION TITLE
Add proxy functions to in setting

### DISCUSSION THREAD

**Discussion by starsang1995** [Tyrrrz/YoutubeDownloader#341] [General]
Hi I'm a user from mainland China. Unfortunately, we have to use VPN-like technologies to access the YouTube services, this would include using YouTube Downloader.

In browsers we are using add-ons like SwitchyOmega to route different domains into different lines for optimized performance, instead of setting the proxy on OS level (not sure if I'm using the correct term as I'm not really familiar with those terms). However, when using YouTube Downloader, since it doesn't support any kind of proxy settings, we have to set the proxy on OS level, and it would cause many other issues accessing other Internet services.

Currently myself and a lot of people I know is using SOCKS as the proxy protocol, so I was wondering if it would be possible to add the SOCKS listening function to YouTube Downloader and it would be easier for Chinese users to use them without needing to set the proxy on OS level?

Any ideas guys?

**Comment by Tyrrrz** [comment]
I believe .NET apps recognize the `HTTP_PROXY` and `HTTPS_PROXY` environment variables. The value should be in the form of `http[s]://[username:password@]<ip_address_or_hostname>:<port>/`. I haven't tested it, but you can try and see if that works.

```bat
set HTTPS_PROXY=http://your-proxy-server:port
start YoutubeDownloader.exe
```
